### ISSUE TITLE
Why do I train my dataset, the output picture is just masked and not fixed, and 300,000 epochs are trained like this?

### ISSUE THREAD

**Issue by R1ckLou** [nipponjo/deepfillv2-pytorch#8]
![iter_500](https://user-images.githubusercontent.com/91020859/187862127-de6907ee-7737-436e-a827-c28bec32ba48.png)
just like this？

**Comment by nipponjo** [comment]
Hello, I think the binary images cause some difficulties. Firstly, I would extend the output range of the generator, so that it can more easily output the -1/1 values. 
You can multiply the tanh activation in `CoarseGenerator` and `FineGenerator` with some value >1 for that.
```python
x = 1.5*self.tanh(x)
```
Also, I think that the L2 loss is more suitable here. In `train.py` you can replace the torch.abs with torch.square.
```python
# G training steps:
losses['ae_loss1'] = config.l1_loss_alpha * \
    torch.mean((torch.square(batch_real - x1)))
losses['ae_loss2'] = config.l1_loss_alpha * \
    torch.mean((torch.square(batch_real - x2)))
```
The LS gan loss can also stabilize training. (in 'configs/train.yaml')
```yaml
gan_loss: 'ls' # options: 'hinge', 'ls'
```
You should see some results quickly. I trained on a similar dataset for ~5k iterations with these changes and that's what I got. Hopefully, the results will still improve with more training :)
![iter_5500](https://user-images.githubusercontent.com/28433296/188019752-0052c9c8-9d77-4f18-bc00-9ca1356c6396.png)
